src/firecracker/src/main.rs

```diff
@@ -75,16 +75,25 @@ impl From<MainError> for ExitCode {
     }
 }
 
-fn main() -> ExitCode {
-    let result = main_exec();
-    if let Err(e) = result {
-        error!("{}", e);
-        e.into()
-    } else {
-        ExitCode::SUCCESS
+/// Exit result wrapper to return specific error codes, log `err.to_string()` and write
+/// `format!({err:?})` to stderr.
+struct MainExit(Result<(), MainError>);
+impl std::process::Termination for MainExit {
+    fn report(self) -> ExitCode {
+        if let Err(err) = self.0 {
+            error!("{err}");
+            eprintln!("Error: {err:?}");
+            ExitCode::from(err)
+        } else {
+            ExitCode::SUCCESS
+        }
```

Comment:
I don't really like adding an additional struct just to print errors. I think you can just modify the main itself:
```rust
fn main() -> ExitCode {
    if let Err(err) = main_exec() {
        error!("{err}");
        eprintln!("Error: {err:?}");
        ExitCode::from(err)
    } else {
        ExitCode::SUCCESS
    }
}

```

Reply:
Fixed.